workflows/pipe-common/scripts/autoscale_sge.py

```diff
@@ -845,9 +840,19 @@ def scale(self):
         additional_hosts = self.host_storage.load_hosts()
         Logger.info('There are %s additional pipelines.' % len(additional_hosts))
         updated_jobs = self.grid_engine.get_jobs()
-        pending_jobs = [job for job in updated_jobs if job.state == GridEngineJobState.PENDING
-                        and self.is_job_fit_or_cancel(job)]
         running_jobs = [job for job in updated_jobs if job.state == GridEngineJobState.RUNNING]
+
+        # kill jobs that are pending and can't be satisfied with requested resource
+        # f.i. we have only 3 instance max and the biggest possible type has 10 cores but job requests 40 cores
+        pending_jobs = []
```

Comment:
Here I would stick with my solution, yes it has a little bit more nesting in the code, but if we move it to method let say 'filter_pending_jobs' it would be pretty accurate and self documented. Also here we have only 2 iteration through the array while filtering instead of 4

Reply:
Sounds reasonable.